src/GafferScene/SceneWriter.cpp

```diff
@@ -53,85 +56,101 @@ using namespace GafferScene;
 namespace
 {
 
-void writeLocation( const GafferScene::ScenePlug *scene, const ScenePlug::ScenePath &scenePath, Context *context, IECore::SceneInterface *output, double time )
+tbb::mutex g_writerMutex;
```

Comment:
As far as I'm aware, it should be safe to write to different SceneCache files concurrently, so this would ideally be a per-file lock rather than a global one. Not the most important thing probably, since I don't think we do any parallel dispatch currently (well, we do it parallel across the farm, but not parallel per process).